components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -709,26 +687,104 @@ private boolean isDomainNameEmbeddedInAttributeValue(String filterOperation, Str
+
+                // Create filter condition for each domain.
+                Condition condition = createConditionForSingleAttributeFilter(domainName, node);
+
+                // Filter users for given condition and domain.
+                String[] userNames = getFilteredUsernames(condition, offset, limit, sortBy, sortOrder,
+                        userStoreDomainName);
+                if (userNames == null) {
+                    userNames = new String[0];
+                }
+
+                // Calculating new offset and limit parameters.
+                int numberOfFilteredUsers = userNames.length;
+                if (numberOfFilteredUsers <= 0 && offset > 1) {
+                    offset = calculateOffset(condition, offset, sortBy, sortOrder, domainName);
+                } else {
+                    // Returned user names size > 0 implies there are users in that domain which is larger than the offset.
+                    offset = 1;
+                    limit = calculateLimit(limit, numberOfFilteredUsers);
+                }
+                filteredUsernames.addAll(Arrays.asList(userNames));
+
+                // If the limit is changed then filtering needs to be stopped.
+                if (limit == 0)
+                    break;
             }
             return filteredUsernames.toArray(new String[0]);
         }
     }
 
+    /**
+     * Method to update the count(limit) when iterating a filter across all domains.
+     *
+     * @param limit                 Counting value (limit)
+     * @param numberOfFilteredUsers Amount of users filtered in the search
+     * @return Calculated new limit (count)
+     */
+    private int calculateLimit(int limit, int numberOfFilteredUsers) {
+        if (log.isDebugEnabled()) {
+            log.debug("Updating the original count to be used for the next domain search. ");
+        }
+        limit = limit - numberOfFilteredUsers;
+        if (limit < 0) {
+            limit = 0;
+        }
+        return limit;
+    }
+
+    /**
+     * Method to update the offset when iterating a filter across all domains.
+     *
+     * @param condition  Condition of the single attribute filter
+     * @param offset     Starting index
+     * @param sortBy     Sort by
+     * @param sortOrder  Sort order
+     * @param domainName Domain to be filtered
+     * @return New calculated offset
+     * @throws CharonException Error while filtering the domain from index 1 to offset
+     */
+    private int calculateOffset(Condition condition, int offset, String sortBy, String sortOrder, String domainName)
+            throws CharonException {
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Filter returned no results for original offset: %d. Checking for number of "
+                    + "matches from the beginning to the original offset: %d for the same filter and updating "
+                    + "the new offset", offset, offset));
```

Comment:
`Filter returned no results for original offset: %d.` : This part should move to where the` calculateOffset` is invoked. Since this debug log is intended to express only the work done by `calculateOffset` method

Reply:
Fixed: f999948935ff2081b9362c4cbe0d6e8f5ee238db